optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/MemberAccessorFactory.java

```diff
@@ -32,7 +34,22 @@
             SolverFactory.class.getSimpleName() + ".create...() method call.";
 
     public static MemberAccessor buildMemberAccessor(Member member, MemberAccessorType memberAccessorType,
-            Class<? extends Annotation> annotationClass) {
+            Class<? extends Annotation> annotationClass,
+            DomainAccessType domainAccessType) {
+        if (domainAccessType == DomainAccessType.GIZMO) {
+            try {
+                // Check if Gizmo on the classpath by verifying we can access one of its classes
+                Class.forName("io.quarkus.gizmo.ClassCreator", false,
+                        Thread.currentThread().getContextClassLoader());
+            } catch (ClassNotFoundException e) {
+                throw new IllegalStateException("When using the domainAccessType (" +
+                        domainAccessType +
+                        ") the classpath or modulepath must contain io.quarkus.gizmo:gizmo.\n" +
+                        "Maybe add a dependency to io.quarkus.gizmo:gizmo.");
+            }
```

Comment:
I see we actually detect if Gizmo is present. So the only thing to make my previous comment happen is the detection of annotations on private fields.